stdio/fprintf.c

```diff
@@ -5,42 +5,81 @@
  *
  * printf.c
  *
- * Copyright 2017 Phoenix Systems
- * Author: Adrian Kepka
+ * Copyright 2017, 2022 Phoenix Systems
+ * Author: Adrian Kepka, Gerard Swiderski
  *
  * This file is part of Phoenix-RTOS.
  *
  * %LICENSE%
  */
 
+#include "errno.h"
 #include "stdio.h"
 #include "stdlib.h"
 #include "unistd.h"
 #include "format.h"
 #include "sys/debug.h"
 
 
-typedef struct _fprintf_ctx_t {
-	FILE *file;
+struct feed_ctx_s {
+	union {
+		FILE *file;
+		int fd;
+	} h;
+	char htype;
```

Comment:
You can use enum type for `htype` - it's ok with both MISRA and our guidelines and will improve code readability